File: backend/schemas/audio.py
from pydantic import BaseModel


class AudioUploadResponse(BaseModel):
    upload_id: str
    filename: str
    duration_s: float
    sample_rate: int


class ProcessRequest(BaseModel):
    upload_id: str
    rir_id: str
    eq_profile_id: str | None = None


class AudioResultResponse(BaseModel):
    job_id: str
    status: str
    result_url: str | None = None
    duration_s: float | None = None

File: frontend/src/types/audio.ts
export interface AudioUploadResponse {
  upload_id: string;
  filename: string;
  duration_s: number;
  sample_rate: number;
}

export interface AudioResultResponse {
  job_id: string;
  status: string;
  result_url?: string;
  duration_s?: number;
}

export type ProcessingState = "idle" | "uploading" | "processing" | "done" | "error";
